resources/build/test/locate_emscripten.tests.sh

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env bash
+# Keyman is copyright (C) SIL Global. MIT License.
+#
+# Unit tests for locate_emscripten.inc.sh
+
+shopt -s inherit_errexit
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
+. "${THIS_SCRIPT%/*}/../../../resources/build/build-utils.sh"
+# END STANDARD BUILD SCRIPT INCLUDE
+
+# shellcheck disable=SC2154
+. "${KEYMAN_ROOT}/resources/locate_emscripten.inc.sh"
+. "${KEYMAN_ROOT}/resources/build/test/testing-framework.inc.sh"
+
+# Mocks
+
+function mocked_builder_die() {
+  echo "DIE: $*" 2>&1
+  exit 99
+}
+
+# Setup/Teardown
+
+function setup_file() {
+  BASE_DIR=$(mktemp -d)
+}
+
+function setup() {
+  mkdir -p "${BASE_DIR}"
+  create_mock  builder_die  mocked_builder_die
+  create_mock  verify_emscripten_version
+}
+
+function teardown() {
+  [[ -v EMSCRIPTEN_BASE ]] && rm -rf "${EMSCRIPTEN_BASE}"
+  [[ -v EMCC ]] && rm -f "${EMCC}"
+  unset EMSCRIPTEN_BASE
+  unset EMCC
+  rm -rf "${BASE_DIR:-}"
+  unmock builder_die
+  unmock verify_emscripten_version
+}
+
+# Tests
+
+function test_locate_emscripten__with_EMSCRIPTEN_BASE_and_executable() {
```

Comment:
For all of these tests, is anything done to clear out pre-existing environment variables that may already have valid `emcc` paths, setup, or similar in place?  It feels like these tests _assume_ a clear environment, but is that actually reasonable?